urdf/panther_macro.urdf.xacro

```diff
@@ -0,0 +1,115 @@
+                       imu_pos_z:=0.083
+                       imu_rot_r:=3.14 
+                       imu_rot_p:=-1.57
+                       imu_rot_y:=0.0
+                       wheel_config_path:=None
+                       simulation_engine:='gazebo-classic'">
+
+    <xacro:if value="${wheel_config_path == 'None'}">
+      <xacro:property name="wheel_config_file" value="$(find panther_description)/config/WH01.yaml"  />
+    </xacro:if>
+    <xacro:unless value="${wheel_config_path == 'None'}">
+      <xacro:property name="wheel_config_file" value="$(arg wheel_config_path)"  />
+    </xacro:unless>
+
+    <xacro:property name="wheel_config" value="${xacro.load_yaml(wheel_config_file)}" />
+
+    <!-- INCLUDE ROBOT PARTS DEFINITIONS -->
+    <xacro:include filename="$(find panther_description)/urdf/body.urdf.xacro" ns="body" />
+    <xacro:include filename="$(find panther_description)/urdf/wheel.urdf.xacro" ns="wheel" />
+
+    <!-- BODY DECLARATION -->
+    <xacro:body.body 
+      wheel_radius="${wheel_config['wheel_radius']}"
+      imu_pos_x="${imu_pos_x}"
+      imu_pos_y="${imu_pos_y}"
+      imu_pos_z="${imu_pos_z}"
+      imu_rot_r="${imu_rot_r}"
+      imu_rot_p="${imu_rot_p}"
+      imu_rot_y="${imu_rot_y}"
+      simulation_engine="${simulation_engine}" />
+
+    <xacro:property name="wheel_separation_x" value="0.44" />
+
+    <!-- WHEEL DECLARATION -->
+    <xacro:wheel.wheel
+      wheel_config="${wheel_config}"
+      wheel_separation_x="${wheel_separation_x}"
+      prefix="fl" />
+
+    <xacro:wheel.wheel 
+      wheel_config="${wheel_config}"
+      wheel_separation_x="${wheel_separation_x}"
+      prefix="fr" />
+      
+    <xacro:wheel.wheel 
+      wheel_config="${wheel_config}"
+      wheel_separation_x="${wheel_separation_x}"
+      prefix="rl" />
+      
+    <xacro:wheel.wheel 
+      wheel_config="${wheel_config}"
+      wheel_separation_x="${wheel_separation_x}"
+      prefix="rr" />
+      
+    
+    <xacro:if value="${simulation_engine == 'gazebo-classic'}">
+      <gazebo>
+        <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
+          <robotNamespace>/</robotNamespace>
+          <legacyModeNS>true</legacyModeNS>
+        </plugin>
+      </gazebo>
+    
+
+      <xacro:if value="${wheel_config['kinematics'] == 'differential'}">
+        <gazebo>
+          <plugin name='skid_steer_drive_controller' filename='libgazebo_ros_skid_steer_drive.so'>
+            <updateRate>20.0</updateRate>
+            <robotBaseFrame>base_link</robotBaseFrame>
+            <wheelSeparation>${wheel_config['wheel_separation']}</wheelSeparation>
+            <wheelDiameter>${wheel_config['wheel_radius']*2.0}</wheelDiameter>
+            <torque>34.5</torque>
+            <leftFrontJoint>fl_joint</leftFrontJoint>
+            <rightFrontJoint>fr_joint</rightFrontJoint>
+            <leftRearJoint>rl_joint</leftRearJoint>
+            <rightRearJoint>rr_joint</rightRearJoint>
```

Comment:
Sorry that I missed it in ROS2 PR, but wouldn't it be better to add `wheel` to joint name (e.g. `fl_wheel_joint`)? I think that  only position prefix doesn't necessarily tell that it will be a wheel.